### PR TITLE
chore: add a pg

### PR DESCRIPTION
**Pull request by juliusmarminge**
add a playground (one for v6, one for v7) with some mixing of utapi and upload button useful for some internal testing since it interacts with both ingeest and api

CleanShot 2024-11-07 at 16 36 50

## Release Notes

- **New Features**
  - Introduced a file upload functionality with user authentication.
  - Added a new `Uploader` component for handling file uploads.
  - Implemented a `FileCard` component for displaying file details with open and delete options.
  - Created a `Skeleton` component for loading states.
  - Added navigation and user authentication in the app layout.
  - Introduced a `Button` component for customizable button styles.
  - Launched a new page component for displaying a list of files.

- **Documentation**
  - Added an example environment configuration file for developers.
  - Created a README file for the "UploadThing Next.js Playground" application.

- **Configuration**
  - New configuration files for TypeScript and Next.js settings.
  - Included the `playground` directory in the workspace management.

### CODE REVIEW

File: playground/app/layout.tsx [hunk 1]
```diff
@@ -0,0 +1,96 @@
...
+async function UTSSR() {
+  await connection();
+
+  return <NextSSRPlugin routerConfig={extractRouterConfig(uploadRouter)} />;
+}
+
+function Nav() {
+  const token = JSON.parse(
+    Buffer.from(process.env.UPLOADTHING_TOKEN!, "base64").toString("utf-8"),
+  ) as { appId: string; regions: string[]; apiKey: string };
```
[the remark this answers is not in the document]
Comment (juliusmarminge) [reply playground/app/layout.tsx]: do you even know what a server compoennt is rabbit?

File: playground/package.json [hunk 2]
```diff
@@ -0,0 +1,28 @@
+{
+  "name": "next-playground",
+  "private": true,
+  "type": "module",
+  "scripts": {
+    "build": "next build",
+    "dev": "next dev --turbopack --port 4000",
+    "lint": "eslint",
+    "typecheck": "tsc"
+  },
```
[the remark this answers is not in the document]
Comment (juliusmarminge) [reply playground/package.json:10]: no it isnt

File: playground/package.json [hunk 3]
```diff
@@ -0,0 +1,29 @@
+{
+  "name": "next-playground",
+  "private": true,
+  "type": "module",
+  "scripts": {
+    "build": "next build",
+    "dev": "next dev --turbopack --port 4000",
+    "lint": "eslint",
```
[the remark this answers is not in the document]
Comment (juliusmarminge) [reply playground/package.json:8]: eslint v9 doesn't need that...

File: playground/package.json [hunk 4]
```diff
@@ -0,0 +1,29 @@
...
+    "react-dom": "18.3.1",
+    "uploadthing": "workspace:*",
+    "zod": "3.23.8"
+  },
+  "devDependencies": {
+    "@types/node": "^20.14.0",
+    "@types/react": "^18.3.12",
+    "@types/react-dom": "^18.3.1",
+    "eslint": "9.14.0",
+    "typescript-eslint": "8.13.0",
```
[the remark this answers is not in the document]
Comment (juliusmarminge) [reply playground/package.json:26]: how old are you man? you're suggesting a bucnh of stuff that was relevant like 3 major versions ago

File: playground-v6/app/layout.tsx [hunk 5]
```diff
@@ -0,0 +1,88 @@
...
+
+export default function RootLayout({
+  children,
+}: {
+  children: React.ReactNode;
+}) {
+  return (
+    <html lang="en">
+      <head>
+        <script src="https://cdn.tailwindcss.com"></script>
```
[the remark this answers is not in the document]
Comment (markflorkowski) [reply playground-v6/app/layout.tsx:21]: @coderabbitai this is a playground for testing purposes, not meant for production deployment

File: playground-v6/components/uploader.tsx [hunk 6]
```diff
@@ -0,0 +1,35 @@
...
+export function Uploader() {
+  const router = useRouter();
+
+  return (
+    <UTButton
+      endpoint="anything"
+      input={{}}
+      onUploadError={(error) => {
+        window.alert(error.message);
+      }}
```
[the remark this answers is not in the document]
Comment (markflorkowski) [reply playground-v6/components/uploader.tsx:21]: @coderabbitai This is a testing playground, no need for additional complexity.

File: playground-v6/lib/actions.ts [hunk 7]
```diff
@@ -0,0 +1,66 @@
...
+  };
+  const exp = session.iat + 60 * 60 * 1000;
+  const cookie = JSON.stringify(session);
+
+  (await cookies()).set(SESSION_COOKIE_NAME, cookie, {
+    httpOnly: true,
+    sameSite: "lax",
+    expires: exp,
+  });
+}
```
[the remark this answers is not in the document]
Comment (markflorkowski) [reply playground-v6/lib/actions.ts:28]: @coderabbitai this is a testing playground, no need for real auth logic. This is just a stand in.

File: playground-v6/package.json [hunk 8]
```diff
@@ -0,0 +1,29 @@
+{
+  "name": "next-playground-v6",
+  "private": true,
+  "type": "module",
+  "scripts": {
+    "build": "next build",
+    "dev": "next dev --turbopack --port 4001",
+    "lint": "eslint",
+    "typecheck": "tsc"
+  },
```
[the remark this answers is not in the document]
Comment (markflorkowski) [reply playground-v6/package.json:10]: 1. Eslint will default to using `.` 
2. This is a testing playground and should never be deployed to production
3. We accept the instability of turbopack

File: playground-v6/package.json [hunk 9]
```diff
@@ -0,0 +1,29 @@
...
+    "zod": "3.23.8"
+  },
+  "devDependencies": {
+    "@types/node": "^20.14.0",
+    "@types/react": "^18.3.12",
+    "@types/react-dom": "^18.3.1",
+    "eslint": "9.14.0",
+    "typescript-eslint": "8.13.0",
+    "typescript": "5.6.3"
+  }
```
[the remark this answers is not in the document]
Comment (markflorkowski) [reply playground-v6/package.json:28]: Current stable is 9.x, 8.56 is EOL

**Review by markflorkowski — COMMENTED** [review]
Lgtm. Looks like Rabbit had a few decent suggestions on polish (empty state, error handling, etc) but definitely not required for a test playground

**Pull request merged**